[2025-05-27T18:07:29.135+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: walmart_etl_dag.generate_synthetic_data manual__2025-05-27T18:07:28+00:00 [queued]>
[2025-05-27T18:07:29.141+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: walmart_etl_dag.generate_synthetic_data manual__2025-05-27T18:07:28+00:00 [queued]>
[2025-05-27T18:07:29.141+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 2
[2025-05-27T18:07:29.147+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): generate_synthetic_data> on 2025-05-27 18:07:28+00:00
[2025-05-27T18:07:29.152+0000] {standard_task_runner.py:57} INFO - Started process 2885 to run task
[2025-05-27T18:07:29.157+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'walmart_etl_dag', 'generate_synthetic_data', 'manual__2025-05-27T18:07:28+00:00', '--job-id', '221', '--raw', '--subdir', 'DAGS_FOLDER/walmart_etl_dag.py', '--cfg-path', '/tmp/tmpsynfitx2']
[2025-05-27T18:07:29.161+0000] {standard_task_runner.py:85} INFO - Job 221: Subtask generate_synthetic_data
[2025-05-27T18:07:29.199+0000] {task_command.py:416} INFO - Running <TaskInstance: walmart_etl_dag.generate_synthetic_data manual__2025-05-27T18:07:28+00:00 [running]> on host a9df2a99636e
[2025-05-27T18:07:29.442+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='walmart_etl_dag' AIRFLOW_CTX_TASK_ID='generate_synthetic_data' AIRFLOW_CTX_EXECUTION_DATE='2025-05-27T18:07:28+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-27T18:07:28+00:00'
[2025-05-27T18:07:29.443+0000] {logging_mixin.py:154} INFO - Starting synthetic data generation process...
[2025-05-27T18:07:29.443+0000] {logging_mixin.py:154} INFO - Generating supplier data...
[2025-05-27T18:07:29.459+0000] {logging_mixin.py:154} INFO - Loaded 50 supplier records to ods_supplier
[2025-05-27T18:07:29.459+0000] {logging_mixin.py:154} INFO - Generating return reason data...
[2025-05-27T18:07:29.462+0000] {logging_mixin.py:154} INFO - Loaded 10 return reason records to ods_return_reason
[2025-05-27T18:07:29.462+0000] {logging_mixin.py:154} INFO - Retrieving existing data...
[2025-05-27T18:07:29.464+0000] {logging_mixin.py:154} INFO - Generating inventory data...
[2025-05-27T18:07:29.464+0000] {logging_mixin.py:154} INFO - No products or stores found. Cannot generate inventory data.
[2025-05-27T18:07:29.464+0000] {logging_mixin.py:154} INFO - Generating returns data...
[2025-05-27T18:07:29.465+0000] {logging_mixin.py:154} INFO - Missing required data for returns generation.
[2025-05-27T18:07:29.465+0000] {logging_mixin.py:154} INFO - Transforming and loading data to staging tables...
[2025-05-27T18:07:29.467+0000] {logging_mixin.py:154} INFO - Supplier and return reason data successfully loaded to staging tables with batch ID: BATCH_SYNTHETIC_20250527180729
[2025-05-27T18:07:29.467+0000] {logging_mixin.py:154} INFO - Starting load to target layer for batch BATCH_SYNTHETIC_20250527180729...
[2025-05-27T18:07:29.473+0000] {logging_mixin.py:154} INFO - Supplier and return reason data successfully loaded to target tables
[2025-05-27T18:07:29.473+0000] {logging_mixin.py:154} INFO - Synthetic data generation and loading complete!
[2025-05-27T18:07:29.473+0000] {python.py:194} INFO - Done. Returned value was: None
[2025-05-27T18:07:29.478+0000] {taskinstance.py:1400} INFO - Marking task as SUCCESS. dag_id=walmart_etl_dag, task_id=generate_synthetic_data, execution_date=20250527T180728, start_date=20250527T180729, end_date=20250527T180729
[2025-05-27T18:07:29.510+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2025-05-27T18:07:29.527+0000] {taskinstance.py:2778} INFO - 1 downstream tasks scheduled from follow-on schedule check
